pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -834,69 +837,89 @@ func (r *ReconcileClusterDeployment) statusUpdate(cd *hivev1.ClusterDeployment,
 	return err
 }
 
-func (r *ReconcileClusterDeployment) resolveInstallerImage(cd *hivev1.ClusterDeployment, imageSet *hivev1.ClusterImageSet, releaseImage string, cdLog log.FieldLogger) (reconcile.Result, error) {
-	// If the .status.clusterVersionsStatus.availableUpdates field is nil,
-	// do a status update to set it to an empty list. All status updates
-	// done by controllers set this automatically. However, the imageset
-	// job does not. If the field is still nil when the imageset job tries
-	// to update the status, then the update will fail validation.
-	if cd.Status.ClusterVersionStatus.AvailableUpdates == nil {
-		return reconcile.Result{}, r.statusUpdate(cd, cdLog)
-	}
-
-	cliImage := images.GetCLIImage()
-	job := imageset.GenerateImageSetJob(cd, releaseImage, controllerutils.ServiceAccountName, imageset.AlwaysPullImage(cliImage))
-	if err := controllerutil.SetControllerReference(cd, job, r.scheme); err != nil {
-		cdLog.WithError(err).Error("error setting controller reference on job")
-		return reconcile.Result{}, err
-	}
+func (r *ReconcileClusterDeployment) resolveInstallerImage(cd *hivev1.ClusterDeployment, imageSet *hivev1.ClusterImageSet, releaseImage string, cdLog log.FieldLogger) (*reconcile.Result, error) {
+	areImagesResolved := cd.Status.InstallerImage != nil && cd.Status.CLIImage != nil
 
-	jobName := types.NamespacedName{Name: job.Name, Namespace: job.Namespace}
-	jobLog := cdLog.WithField("job", jobName)
+	jobKey := client.ObjectKey{Namespace: cd.Namespace, Name: imageset.GetImageSetJobName(cd.Name)}
```

Comment:
I do not recall how this specific change came about. I probably re-wrote most of this function, and I have a habit of using `client.ObjectKey` rather than `types.NamespacedName`. I do not think that I would deliberately change an existing use of the latter to replace it with the former. With that being said, there are a few reasons why I prefer to use `client.ObjectKey`.
1) The controller-runtime API ask for a `client.ObjectKey`. All things being equal, I would rather use the term that the API is using.
2) If I use `types.NamespacedName`, then I have to import that package into the file. The `client` package is already imported in almost all cases where I would be using a controller-runtime client.
3) Using the `types.NamespacedName` type leads to variable names like `jobName` which are confusing since we typically use name to refer to just the Name of the object and not the Name+Namespace of the object.